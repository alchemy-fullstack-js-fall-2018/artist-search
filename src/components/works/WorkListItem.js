import React from 'react';
import PropTypes from 'prop-types';
import { ROUTES } from '../../routes/routes';
import { Link } from 'react-router-dom';

const WorkListItem = ({ artist, title }) => {

  return (
    <li>
      <Link to={ROUTES.WORK.linkTo(artist, title)}>{title}</Link>
    </li>
  );

};

WorkListItem.propTypes = {
  artist: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
};

export default WorkListItem;
