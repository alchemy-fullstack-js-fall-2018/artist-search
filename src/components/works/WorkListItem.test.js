import React from 'react';
import { shallow } from 'enzyme';
import WorkListItem from './WorkListItem';

describe ('tests for WorkListItem component', () => {
  const props = {
    artist: 'Joni Mitchell',
    title: 'Blue'
  };
  it('renders an WorkListItem component', () => {
    const wrapper = shallow(<WorkListItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
