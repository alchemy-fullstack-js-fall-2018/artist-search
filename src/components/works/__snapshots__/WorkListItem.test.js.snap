// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for WorkListItem component renders an WorkListItem component 1`] = `
<li>
  <Link
    replace={false}
    to="/works/Joni Mitchell/Blue"
  >
    Blue
  </Link>
</li>
`;
